src/3d/qgscameracontroller.cpp

```diff
@@ -457,6 +457,14 @@ void QgsCameraController::onPositionChangedTerrainNavigation( Qt3DInput::QMouseE
       mDragPointCalculated = true;
     }
 
+    // Make sure we apply grab move when the cursor is inside the view port
+    if ( !mViewport.contains( mouse->x(), mouse->y() ) )
+      return;
```

Comment:
This is very limiting to the user, granted that the 3d views are often small windows/docked widgets.